amaranth-chrome-ext/src/AmaranthUtil.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Utility class with helpful functions for the Amaranth Chrome extension.
+ * All methods are static.
+ */
+class AmaranthUtil {
+  /**
+   * Removes special characters from a string.
+   * @param {string} str String to remove special characters from
+   * @param {string} filters Characters to remove from str
+   * @return {string} `str` with all characters in `filters` removed
+   */
+  static removeSpecialCharacters(
+      str, filters='!"#$%&()*+,-./:;<=>?@[\\]^_`{|}~\t\n') {
+    let newStr = '';
+    for (const char of str) {
```

Comment:
In Java this could be a FluentIterable which would prevent string concatenation in this manner and would prevent allocation memory for a new variable. Is there a similar way to do this in JS?

Reply:
This is a good point, and I had to look into it. Apparently [the V8 runtime optimizes string concatenation fairly well](https://stackoverflow.com/questions/7299010/why-is-string-concatenation-faster-than-array-join/7299040), so this is actually one of the faster methods of doing this. However, it does look like new strings are created each time. I can imagine alternatives using a list of characters which are joined at the end, or a functional solution like this this pseudocode:

```
strArr = str.toArray()
strArr.filter(specialCharacters)
return strArr.toString()
```

But I'm temped to leave it as-is for the sake of readability since it's speed is reasonable.